request_test.go

```diff
@@ -3726,3 +3726,40 @@ func TestRequest_Panics(t *testing.T) {
 		assert.Panics(t, func() { newRequest(newMockChain(t), config, "GET", "") })
 	})
 }
+
+func TestRequest_Reporter(t *testing.T) {
+	reporter := newMockReporter(t)
+
+	config := Config{
+		RequestFactory: DefaultRequestFactory{},
+		Client:         &mockClient{},
+		Reporter:       reporter,
+	}
+
+	req := NewRequestC(config, "GET", "/")
+	req.WithReporter(reporter)
+	req.WithClient(nil)
+	req.chain.assert(t, failure)
+
+	assert.True(t, reporter.reported)
```

Comment:
Can you please rework this test to use two reporters?

* first reporter passed to config
* second passed to WithReporter
* we check that failure was reported to the second, and not to the first